### ISSUE TITLE
&nbsp; gets eaten out for components with long lines

### ISSUE THREAD

**Issue by methyl** [prettier/prettier#2217]
Check out this example:
```js
<div>
  <div>&nbsp;</div>
  <div className="very long class name which will definitely exceed our 80 chars limit">
    &nbsp;
  </div>
</div>
```

which gets converted to
```js
<div>
  <div>&nbsp;</div>
  <div className="very long class name which will definitely exceed our 80 chars limit" />
</div>;
```

`&nbsp;` in second `<div>` is eaten out by prettier, which changes behavior of code.

**Comment by vjeux** [comment]
cc @karl

**Comment by karl** [comment]
Looks like it might not be to do with line length. I can trigger the bug by adding a new line before or after the `&nbsp;`

https://prettier.github.io/prettier/#%7B%22content%22%3A%22%3Cdiv%3E%5Cn%20%20%3Cdiv%3E%26nbsp%3B%3C%2Fdiv%3E%5Cn%20%20%3Cdiv%3E%5Cn%20%20%20%20%26nbsp%3B%5Cn%20%20%3C%2Fdiv%3E%5Cn%3C%2Fdiv%3E%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Afalse%2C%22trailingComma%22%3A%22none%22%2C%22bracketSpacing%22%3Atrue%2C%22jsxBracketSameLine%22%3Afalse%2C%22parser%22%3A%22babylon%22%2C%22semi%22%3Atrue%2C%22useTabs%22%3Afalse%2C%22doc%22%3Afalse%2C%22ast%22%3Afalse%7D%7D

**Comment by methyl** [comment]
You're right, the problem is that when you have long line like
```
  <div className="very long class name which will definitely exceed our 80 chars limit">&nbsp;</div>
```

then it will be converted to 
```
<div className="very long class name which will definitely exceed our 80 chars limit">
  &nbsp;
</div>;
```

and in consequence eaten out in next phase.

**Comment by karl** [comment]
I'm working on a fix for this 👍🏻

**Comment by karl** [comment]
Thanks for the bug report @methyl! 

This was an oversight in our tests. I've created a PR to fix it, so should be fixed for the next release of Prettier.

**Comment by bakkot** [comment]
Fixed by #2220; closing.
